File: OOP September 2014/Lectures/02_02_Exception-Handling/2. Exception-Handling-Demos/Read-Text-File/ReadTextFile.cs
using System;
using System.IO;

class ReadTextFile
{
    static void Main()
    {
        StreamReader reader = new StreamReader(@"..\..\ReadTextFile.cs");
        using (reader)
        {
            int lineNumber = 0;
            string line = reader.ReadLine();
            while (line != null)
            {
                lineNumber++;
                Console.WriteLine("Line {0}: {1}", lineNumber, line);
                line = reader.ReadLine();
            }
        }
    }
}

File: OOP September 2014/Lectures/08_OOP-in-Java/8. OOP-in-Java-Demos/src/ReadTextFile.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class ReadTextFile {
	private static String readTextFile(String fileName) throws IOException {
		StringBuilder lines = new StringBuilder();
		try (BufferedReader reader = 
				new BufferedReader(new FileReader(fileName))) {
			String newLine = System.getProperty("line.separator");
			String line;
			while ((line = reader.readLine()) != null) {
				lines.append(line);
				lines.append(newLine);
			}
		}
		return lines.toString();
	}
	
	public static void main(String[] args) {
		String fileName = "src/ReadTextFile.Java";
		try {
			String fileContent = readTextFile(fileName);
			System.out.println(fileContent);
		} catch (IOException ioex) {
			System.out.println("Can not read file: " + fileName);
		}
	}
}
